prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,39 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if is_color_like(colors) or colors is None or isinstance(colors, Number):
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+    elif isinstance(colors, dict):
+        if labels is None:
+            raise TypeError('color cannot be a dict unless labels are provided')
+        colors_dict = colors
+        colors = [colors_dict[label] for label in labels]
+
+    if isinstance(colors, list):
+        if len(colors) != num:
+            raise ValueError('colors should have the length of the set to be colored or satisfy matplotlib color rules')
+
+        for color in colors:
+            if not is_color_like(color):
+                if not allowNumbers:
+                    raise ValueError('each element of colors should satisfy matplotlib color rules')
+                elif not isinstance(color, Number):
+                    raise ValueError('each element of colors should be a number or satisfy matplotlib color rules')
+
+            if not isinstance(color, type(colors[0])):
+                raise TypeError('each element of colors should have the same type')
```

Comment:
Is this check necessary? Couldn't the matplotlib function handle colors defined in different ways in the same list?

Reply:
oh, I guess it probably could